config.go

```diff
@@ -32,6 +32,14 @@ type Config struct {
 	// falls back to default union resolution behavior based on the value of
 	// UnionResolutionError.
 	PartialUnionTypeResolution bool
+
+	// Disable caching layer for encoders, forcing them to get rebuilt on every
+	// call to Marshal()
+	DisableEncoderCaching bool
+
+	// Disable caching layer for decoders, forcing them to get rebuilt on every
+	// call to Unmarshal()
+	DisableDecoderCaching bool
```

Comment:
This can be a single option `DisableCache`.

Reply:
I considered it but thought that adding the modularity of disabling only 1 side and not the other was worth the added configuration option. If you would rather see only 1 option, I can merge them, I do not mind

Reply:
It should not be a regular use case, so I would prefer a single option.

Reply:
perfect, I will fix